src/main/java/com/google/cloud/tools/gradle/appengine/standard/AppEngineStandardPlugin.java

```diff
@@ -123,11 +119,9 @@ private void configureExtensions() {
                   .resolve("appengine-web.xml")
                   .toFile();
 
-          // configure the deploy extensions's project/version parameters
-          StandardDeployTargetResolver resolver =
-              new StandardDeployTargetResolver(appengineWebXml, cloudSdkOperations.getGcloud());
-          deploy.setProjectId(resolver.getProject(deploy.getProjectId()));
-          deploy.setVersion(resolver.getVersion(deploy.getVersion()));
+          // configure the supplier for project/version parameters
+          deploy.setDeployTargetResolver(
+              new StandardDeployTargetResolver(appengineWebXml, cloudSdkOperations));
```

Comment:
What do you think of saving the `new StandardDeployTargetResolver` instance in a local var, and reusing it below in `runExtension.setDeployTargetResolver()`?